Include/pyintrinsics.h

```diff
@@ -0,0 +1,28 @@
+#ifndef Py_PYINTRINSICS_H
+#define Py_PYINTRINSICS_H
+
+/* Return the smallest integer k such that n < 2**k, or 0 if n == 0.
+ * Equivalent to floor(lg(x))+1.  Also equivalent to: bitwidth_of_type -
+ * count_leading_zero_bits(x)
+ */
+
+#if defined(__GNUC__)
```

Comment:
[gcc 3.4](https://gcc.gnu.org/onlinedocs/gcc-3.4.6/gcc/Keyword-Index.html#Keyword-Index). I made the guard more precise. It also works for clang.

Btw. while looking how this is usually done in Python I found a typo that would affect gcc 3.x: https://github.com/python/cpython/blob/master/Include/pymacro.h#L56 should say `__GNU*C*_MINOR__`. I'll make a seperate PR.